CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+## Summary of upstream dockerfiles
+
+The kserve server rocks in this repo build images for the upstream dockerfiles located [here](https://github.com/kserve/kserve/tree/master/python).  These server images all have the following common traits:
+
+* they are implemented as python packages and use [poetry](https://python-poetry.org/) to manage their dependencies
+* each server installs its own server-specific package (ex: [sklearn](https://github.com/kserve/kserve/tree/master/python/sklearnserver))
+* they all install a common [kserve](https://github.com/kserve/kserve/tree/master/python/kserve) package
+
+The Dockerfile for each of these images takes advantage of how each server is defined as a poetry package, using `poetry install` in the Dockerfile directly.
+
+## Implementation details of the ROCKs in this repo
+
+The ROCKs for the kserve servers require some atypical workarounds, mostly due to the upstream project using poetry to install its dependencies.  These are documented here in detail, and briefly noted in the rockcraft.yaml files in this repository.
+
+### Installing Python/pip via overlay-packages
+
+We use overlay-packages to get a python3.10 that we can `pip install` packages into.  For some reason if we add python3.10 as a build-package, the python packages we `pip install`/`poetry install` are not added to this python but to a different one.  This problem will first show up when calling `pip install poetry; poetry config ...`, where `pip install poetry` will work fine buy `poetry config ...` will return with command not found.  
```

Comment:
I assume that we don't know the rockcraft reason behind it, right? Is there a chance that python `pip` is used instead python3 `pip` and we would need `python-is-python3` package?

Reply:
I 've used that [here](https://github.com/canonical/seldonio-rocks/blob/main/mlserver-huggingface/rockcraft.yaml#L38) (although that's not clear what this means for python in the final rock) after discussion with rockcraft in mattermost [here](https://chat.charmhub.io/charmhub/pl/6y9qic1qpjdkfnc8bdg1en5fue).

Reply:
`python-is-python3` can actually replace the symlink fix, but I can't get it to work as a replacement to the overlay-packages.  I might not be doing it quite right though.  

What do you say to keeping it as is for now in this doc, but then I'll test `python-is-python3` with the server images and apply a patch to them separately?

Reply:
Sure. Let's do it this way. And if this works, we can update this doc too.